cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -68,5 +69,98 @@ func (a *Actuator) Update(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 // Exists test for the existance of a machine and is invoked by the Machine Controller
 func (a *Actuator) Exists(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (bool, error) {
 	glog.Infof("Checking if machine %v for cluster %v exists.", machine.Name, cluster.Name)
-	return libvirtutils.DomainExists(machine)
+
+	// decode config
+	machineProviderConfig, err := machineProviderConfigFromClusterAPIMachineSpec(&machine.Spec)
+	if err != nil {
+		return false, fmt.Errorf("error getting machineProviderConfig from spec: %v", err)
+	}
+
+	// build libvirtd client
+	client, err := libvirtutils.BuildClient(machineProviderConfig.URI)
+	if err != nil {
+		return false, fmt.Errorf("Failed to build libvirt client: %s", err)
+	}
+	return libvirtutils.DomainExists(machine.Name, client)
+}
+
+// CreateVolumeAndMachine creates a volume and domain which consumes the former one
+func createVolumeAndDomain(machine *clusterv1.Machine, offset int) error {
```

Comment:
nit: this function does too much. Why not two separate functions that can be combined?